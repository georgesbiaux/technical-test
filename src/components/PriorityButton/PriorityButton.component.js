import React from 'react';
import StyledPriorityButton from './PriorityButton.style';

const PriorityButton = props => (
  <StyledPriorityButton className="priority-button" priority={props.priority}>
    <i className="fa fa-bars" />
  </StyledPriorityButton>
);

export default PriorityButton;
