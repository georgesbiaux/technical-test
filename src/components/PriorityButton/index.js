export { default } from './PriorityButton.component';
