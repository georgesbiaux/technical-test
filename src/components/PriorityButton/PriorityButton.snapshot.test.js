import React from 'react';
import { render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PriorityButton from './PriorityButton.component';

describe('<PriorityButton>', () => {
  it('should render correctly when priority is false', () => {
    const props = {
      priority: false,
    };
    const tree = render(<PriorityButton {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('should render correctly when priority is true', () => {
    const props = {
      priority: true,
    };
    const tree = render(<PriorityButton {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
