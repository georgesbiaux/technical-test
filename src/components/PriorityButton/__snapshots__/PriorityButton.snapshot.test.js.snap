// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriorityButton> should render correctly when priority is false 1`] = `
<button
  class="priority-button sc-bdVaJa kWSmXR"
>
  <i
    class="fa fa-bars"
  >
    
  </i>
</button>
`;

exports[`<PriorityButton> should render correctly when priority is true 1`] = `
<button
  class="priority-button sc-bdVaJa brwQhq"
>
  <i
    class="fa fa-bars"
  >
    
  </i>
</button>
`;
